File: 357/Main.java
import java.io.*;

public class Main {
   final static int WAYS = 6001;

   public static void main(String args[]) throws IOException {
      int[] coin = {5, 10, 25, 50};
      long[] way = new long[WAYS];
      String line;
      StringBuilder answer = new StringBuilder();
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

      for (int i = 0; i < way.length; i++) {
         way[i] = 1;
      }

      for (int i = 0; i < coin.length; i++) {
         for (int j = (coin[i] / 5); j < way.length; j++) {
            way[j] += way[j - (coin[i] / 5)];
         }
      }

      while ((line = in.readLine()) != null) {
         int cents = Integer.parseInt(line);
         if (cents < 5) {
            answer.append("There is only 1 way to produce ");
         } else {
            answer.append("There are ");
            answer.append(Long.toString(way[cents / 5]));
            answer.append(" ways to produce ");
         }
         answer.append(Integer.toString(cents));
         answer.append(" cents change.\n");
      }
      System.out.print(answer);
   }
}

File: 357/main.c
#include <stdio.h>

#define MAX_WAYS 6001

int main(int argc, char* argv[])
{
   int i, j, cents;
   int coin[4] = {5, 10, 25, 50};
   long long int way[MAX_WAYS];

   for (i = 0; i < MAX_WAYS; i++)
      way[i] = 1;

   for (i = 0; i < 4; i++) {
      for (j = (coin[i] / 5); j < MAX_WAYS; j++) {
         way[j] += way[j - (coin[i] / 5)];
      }
   }

   while (scanf("%d", &cents) == 1) {
      if (cents < 5) {
         printf("There is only 1 way to produce %d cents change.\n", cents);
      } else {
         printf("There are %lld ways to produce %d cents change.\n", way[cents / 5], cents);
      }
   }

   return 0;
}
